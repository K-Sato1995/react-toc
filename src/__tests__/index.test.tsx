import * as React from 'react'
import Toc from '../index'
import renderer from 'react-test-renderer'

const markdownText = `
# Heading1
This is a test sentence.

## Heading2
This is a test sentence.

### Heading3
This is a test sentence.
`

describe('<Toc/>', () => {
  it('returns null if the markdownText dose not exist', () => {
    const component = renderer.create(<Toc markdownText={''} />)
    const tree = component.toJSON()

    expect(tree).toBe(null)
  })

  it('renders properly', () => {
    const component = renderer.create(<Toc markdownText={markdownText} />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders properly with titleLimit option', () => {
    const component = renderer.create(
      <Toc markdownText={markdownText} titleLimit={3} />,
    )
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders properly with lowestHeadingLevel option', () => {
    const component = renderer.create(
      <Toc markdownText={markdownText} titleLimit={3} lowestHeadingLevel={4} />,
    )
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders properly with className option', () => {
    const component = renderer.create(
      <Toc
        markdownText={markdownText}
        titleLimit={3}
        lowestHeadingLevel={4}
        className={'customClassName'}
      />,
    )
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders properly with type option', () => {
    const component = renderer.create(
      <Toc
        markdownText={markdownText}
        titleLimit={3}
        lowestHeadingLevel={4}
        className={'customClassName'}
        type="raw"
      />,
    )
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
