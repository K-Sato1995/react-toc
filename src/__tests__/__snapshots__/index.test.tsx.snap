// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Toc/> renders properly 1`] = `
<ul
  className={undefined}
>
  <li>
    <a
      href="#heading1!?"
    >
      Heading1!?
      
    </a>
  </li>
  <ul>
    <li>
      <a
        href="#heading2!"
      >
        Heading2!
        
      </a>
    </li>
  </ul>
  <ul>
    <ul>
      <li>
        <a
          href="#heading3?"
        >
          Heading3?
          
        </a>
      </li>
    </ul>
  </ul>
</ul>
`;

exports[`<Toc/> renders properly with className option 1`] = `
<ul
  className="customClassName"
>
  <li>
    <a
      href="#heading1!?"
    >
      Hea..
    </a>
  </li>
  <ul>
    <li>
      <a
        href="#heading2!"
      >
        Hea..
      </a>
    </li>
  </ul>
  <ul>
    <ul>
      <li>
        <a
          href="#heading3?"
        >
          Hea..
        </a>
      </li>
    </ul>
  </ul>
</ul>
`;

exports[`<Toc/> renders properly with customMatchers option 1`] = `
<ul
  className="customClassName"
>
  <li>
    <a
      href="#heading1--"
    >
      Heading1!?
      
    </a>
  </li>
  <ul>
    <li>
      <a
        href="#heading2-"
      >
        Heading2!
        
      </a>
    </li>
  </ul>
  <ul>
    <ul>
      <li>
        <a
          href="#heading3-"
        >
          Heading3?
          
        </a>
      </li>
    </ul>
  </ul>
</ul>
`;

exports[`<Toc/> renders properly with lowestHeadingLevel option 1`] = `
<ul
  className={undefined}
>
  <li>
    <a
      href="#heading1!?"
    >
      Hea..
    </a>
  </li>
  <ul>
    <li>
      <a
        href="#heading2!"
      >
        Hea..
      </a>
    </li>
  </ul>
  <ul>
    <ul>
      <li>
        <a
          href="#heading3?"
        >
          Hea..
        </a>
      </li>
    </ul>
  </ul>
</ul>
`;

exports[`<Toc/> renders properly with titleLimit option 1`] = `
<ul
  className={undefined}
>
  <li>
    <a
      href="#heading1!?"
    >
      Hea..
    </a>
  </li>
  <ul>
    <li>
      <a
        href="#heading2!"
      >
        Hea..
      </a>
    </li>
  </ul>
  <ul>
    <ul>
      <li>
        <a
          href="#heading3?"
        >
          Hea..
        </a>
      </li>
    </ul>
  </ul>
</ul>
`;

exports[`<Toc/> renders properly with type option 1`] = `
<ul
  className="customClassName"
>
  <li>
    <a
      href="#heading1!?"
    >
      Hea..
    </a>
  </li>
  <ul>
    <li>
      <a
        href="#heading2!"
      >
        Hea..
      </a>
    </li>
  </ul>
  <ul>
    <ul>
      <li>
        <a
          href="#heading3?"
        >
          Hea..
        </a>
      </li>
    </ul>
  </ul>
</ul>
`;
